cmake/Modules/FindCBLAS.cmake

```diff
@@ -5,27 +5,56 @@
 #  CBLAS_LIBRARIES   - List of libraries when using CBLAS.
 #  CBLAS_FOUND       - True if CBLAS found.
 
-include (FindPackageHandleStandardArgs)
-
-if (APPLE)
-  find_path (CBLAS_INCLUDES_PRE Accelerate.h)
-  find_library (CBLAS_LIBRARIES_PRE NAMES Accelerate)
-else ()
-  find_path (CBLAS_INCLUDES_PRE cblas.h)
-  find_library (CBLAS_LIBRARIES_PRE NAMES blas)
-endif ()
-
-# handle the QUIETLY and REQUIRED arguments and set CBLAS_FOUND to TRUE if
-# all listed variables are TRUE
-
-if (CBLAS_LIBRARIES_PRE)
-  set (CBLAS_LIBRARIES ${CBLAS_LIBRARIES_PRE})
-endif ()
-if (CBLAS_INCLUDES_PRE)
-  set (CBLAS_INCLUDES ${CBLAS_INCLUDES_PRE})
-  set (CBLAS_FOUND TRUE CACHE INTERNAL "")
-endif ()
+include(FindPackageHandleStandardArgs)
+include(CheckSymbolExists) 
 
-find_package_handle_standard_args (CBLAS DEFAULT_MSG CBLAS_LIBRARIES CBLAS_INCLUDES)
 
+function(FIND_AND_CHECK_CBLAS _libname _includename _symbol)
+  find_path(_CBLAS_INCLUDES ${_includename})
+  find_library(_CBLAS_LIBRARIES NAMES ${_libname})
+
+  # check if cblas symbol is present
+  if(_CBLAS_LIBRARIES AND _CBLAS_INCLUDES) 
+    set(CMAKE_REQUIRED_INCLUDES ${_CBLAS_INCLUDES})
+    set(CMAKE_REQUIRED_LIBRARIES ${_CBLAS_LIBRARIES})
+    check_symbol_exists(${_symbol} ${_includename} _HAVE_CBLAS_SYMBOL)
```

Comment:
That's a bummer. I see that your suggestion is a simple solution, but unfortunately it only checks the include file, so it would not detect the particular issue I ran into (the correct headers were found, but the wrong library). I will investigate some more.